swap_meet/vendor.py

```diff
@@ -1,2 +1,79 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        self.inventory = inventory if inventory else []
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+        else:
+            self.inventory.remove(item)
+            return item
+
+    def get_by_id(self,id):
+        for item in self.inventory:
+            if item.id == id:
+                return item
+        return None
+
+    def swap_items(self, other_vendor, my_item, their_item):
+        vendor_item = other_vendor.get_by_id(their_item.id)
+        an_item = self.get_by_id(my_item.id)
+
+        if an_item is not None and vendor_item is not None:
+            other_vendor.add(my_item)
+            self.add(their_item)
+            other_vendor.remove(their_item)
+            self.remove(my_item)
+            return True
+        else:
+            return False
+
```

Comment:
The main thing here is that `my_item` and `their_item` will either exist in their respective lists or not! I like the idea of using the `get_by_id` method you created. However, lines 23-24 are either just creating a second reference to each item or getting `None`, which is a little redundant. 

Rather than creating a second reference to each item, it might behoove you to simply see if each item exists in their respective lists. If either one is missing, go ahead and return False! Then you can proceed with lines 27-31 outside the conditional. This avoids some redundancy and also more closely follows the guard clause/ rest of the code pattern we often use.